src/main/minio.js

```diff
@@ -470,6 +487,14 @@ export class Client {
 
     var payload = ''
 
+    // Region already set in constructor, validate if
+    // caller requested bucket location is same.
+    if (this.region) {
+      if (region && region !== this.region) {
```

Comment:
Looks like the API doesn't accept null region (see isString(region) above), I think it is good to let the user pass null, and then we can use this.region.